scene/gui/color_picker.cpp

```diff
@@ -175,31 +175,18 @@ void ColorPicker::_update_color() {
 	updating = false;
 }
 
-void ColorPicker::_update_presets() {
+void ColorPicker::_preset_draw() {
 	Size2 size = bt_add_preset->get_size();
-	preset->set_custom_minimum_size(Size2(size.width * presets.size(), size.height));
-
-	PoolVector<uint8_t> img;
-	img.resize(size.x * presets.size() * size.y * 3);
-
-	{
-		PoolVector<uint8_t>::Write w = img.write();
-		for (int y = 0; y < size.y; y++) {
-			for (int x = 0; x < size.x * presets.size(); x++) {
-				int ofs = (y * (size.x * presets.size()) + x) * 3;
-				w[ofs + 0] = uint8_t(CLAMP(presets[(int)x / size.x].r * 255.0, 0, 255));
-				w[ofs + 1] = uint8_t(CLAMP(presets[(int)x / size.x].g * 255.0, 0, 255));
-				w[ofs + 2] = uint8_t(CLAMP(presets[(int)x / size.x].b * 255.0, 0, 255));
-			}
-		}
-	}
+	Rect2 full_preset = Rect2(Point2(), Size2(size.width * presets.size(), size.height));
+	preset->set_custom_minimum_size(full_preset.size);
 
-	Ref<Image> i = memnew(Image(size.x * presets.size(), size.y, false, Image::FORMAT_RGB8, img));
+	preset->draw_texture_rect(get_icon("MiniCheckerboard", "EditorIcons"), full_preset, true);
 
-	Ref<ImageTexture> t;
-	t.instance();
-	t->create_from_image(i);
-	preset->set_texture(t);
+	Rect2 preset_rect = Rect2(Point2(), size);
+	for(int i = 0; i < presets.size(); i++) {
```

Comment:
Clang-format requires a space after for.
`for (int i = 0; i < presets.size(); i++) {`